### PR TITLE
Fix for PH5 validate incorrectly marking stations as missing data

### PR DESCRIPTION
**Pull request by nick-iris**
### What does this PR do?

Provides a fix for `ph5_validate` incorrectly identifying stations as missing data as well as crashing from opening too many files.

### Relevant Issues?
Issue #308 

### Checklist
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [x] All tests pass.
- [x] Any new or changed features have are documented.
- [ ] Changes have been added to `CHANGELOG.txt` .
- [x] First time contributors have added your name to `CONTRIBUTORS.txt` .

### CODE REVIEW

**Comment by nick-iris** [comment]
This bug seems to have existed in the last release (v4.1.2) as well.

**Comment by dsentinel** [comment]
I'm not familiar with the technical details, but some general comments...

File: ph5/utilities/ph5validate.py [hunk 1]
```diff
@@ -474,12 +473,13 @@ def check_station_completeness(self, station):
...
-                         "You may need to reload the raw "
-                         "data for this station."
-                         .format(str(station_id)))
         try:
+            self.ph5.read_das_t(das_serial, reread=False)
+            if das_serial not in self.ph5.Das_t:
+                error.append("No data found for das serial number {0}. "
+                             "You may need to reload the raw "
+                             "data for this station."
+                             .format(str(das_serial)))
```
Comment (dsentinel) [inline ph5/utilities/ph5validate.py]: Perhaps this should be in it's own try.
With this and the original try having specific exceptions expected.

File: ph5/utilities/ph5validate.py [hunk 2]
```diff
@@ -494,7 +494,12 @@ def check_station_completeness(self, station):
                 time = int(true_pickup - pickup_time)
                 warning.append("Data exists after pickup time: "
                                + str(time) + " seconds ")
+            self.ph5.forget_das_t(das_serial)
         except BaseException:
```
Comment (dsentinel) [inline ph5/utilities/ph5validate.py]: Is there a reason `Exception` isn't used, or better yet the possible specific failures?

Comment (nick-iris) [reply ph5/utilities/ph5validate.py]: Yes this should be `Exception`. I'll change it.

**Comment by nick-iris** [comment]
> How can validate open too many files?
> The ph5 is a master and perhaps 10's of mini files. Are others being accessed?

In the case of Wavefields (16-015 / YW 2016) there are 278 miniPH5 files. As you can see from the stack trace I posted in issue https://github.com/PIC-IRIS/PH5/issues/308, the HDF5 library raised an exception because there were too many open files.

PH5 would open the miniPH5 files and then never close them. So I added a call to the ph5api `forget_das_t` method to ensure that they get closed.

**Comment by nick-iris** [comment]
Even 278 files doesn't seem like it should put it over the edge though, so perhaps there is something else going on within the HDF5 library that is causing this issue.

**Comment by nick-iris** [comment]
@dsentinel Please approve the PR if you are satisfied with the changes.

**Pull request merged**